account_bank_statement_import_coda/wizard/account_bank_statement_import_coda.py

```diff
@@ -0,0 +1,196 @@
+    def get_st_line_note(self, line, information_dict):
+        """This method returns a formatted note from line information
+        """
+        note = []
+        if line.counterparty_name:
+            note.append(_('Counter Party') + ': ' +
+                        line.counterparty_name)
+        if line.counterparty_number:
+            note.append(_('Counter Party Account') + ': ' +
+                        line.counterparty_number)
+        if line.counterparty_address:
+            note.append(_('Counter Party Address') + ': ' +
+                        line.counterparty_address)
+        infos = information_dict.get(line.transaction_ref, [])
+        if line.communication or infos:
+            communications = []
+            if line.communication:
+                communications.append(line.communication)
+            for info in infos:
+                communications.append(info.communication)
+            note.append(_('Communication') + ': ' +
+                        " ".join(communications))
+        return note and '\n'.join(note) or None
+
+    def get_st_line_name(self, line, globalisation_dict):
+        """
+        This method must return a valid name for the statement line
+        The name is the statement communication if exists or
+        the communication of the related globalisation line if exists or
+        '/'
+        """
+        name = line.communication
+        if not name and line.ref_move in globalisation_dict:
+            name = globalisation_dict[line.ref_move].communication
+        return name or '/'
+
+    def get_st_line_vals(self, line, globalisation_dict, information_dict):
+        """
+        This method must return a dict of vals that can be passed to create
+        method of statement line in order to record it. It is the
+        responsibility of every parser to give this dict of vals,
+        so each one can implement his own way of recording the lines.
+            :param:  line: a dict of vals that represent a line of
+            result_row_list
+            :return: dict of values to give to the create method of
+            statement line,
+                     it MUST contain at least:
+                {
+                    'name':value,
+                    'date':value,
+                    'amount':value,
+                    'ref':value,
+                }
+        """
+        amount = line.transaction_amount
+        if line.transaction_amount_sign == AmountSign.DEBIT:
+            amount = - amount
+        return {
+            'name': self.get_st_line_name(line, globalisation_dict),
+            'date': line.entry_date or datetime.datetime.now().date(),
+            'amount': amount,
+            'ref': line.ref,
+            'partner_name': line.counterparty_name or None,
+            'account_number': line.counterparty_number or None,
+            'note': self.get_st_line_note(line, information_dict),
+            'unique_import_id': line.ref + line.transaction_ref,
```

Comment:
we get a problem here testing it with Triodos Coda.
the uniqe_import_id is too weak and led to ignore statement and or statement line import.
in our implementation we build the uniqe_import_id  like this statement.coda_seq_number + '-' + statement.old_balance_date + '-' + statement.new_balance_date + '-' + move.ref

we also found that trasanction_ref returned from pycoda is in fact the bank reference from which the payment as been made. This reference is not mandatory and can be empty(N° de référence de la banque. Cette information est strictement informative)